DiverseSelector/test/test_ClusteringBased.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# The DiverseSelector library provides a set of tools to select molecule
+# subset with maximum molecular diversity.
+#
+# Copyright (C) 2022 The QC-Devs Community
+#
+# This file is part of DiverseSelector.
+#
+# DiverseSelector is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# DiverseSelector is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+#
+# --
+
+"""Testing for the dissimilarity-based selection algorithms."""
+
+from DiverseSelector import clustering_based
+from DiverseSelector.test.common import generate_synthetic_data
+import numpy as np
+
+coords, class_labels, arr_dist = generate_synthetic_data(n_samples=100,
+                                                         n_features=2,
+                                                         n_clusters=3,
+                                                         pairwise_dist=True,
+                                                         metric="euclidean",
+                                                         random_state=42)
+
+
+def test_kmeans():
+    """Testing the kmeans selection algorithm."""
+    selector = clustering_based.ClusteringSelection(num_selected=12,
+                                                    num_clusters=3,
+                                                    clustering_method="k-means",
+                                                    metric='euclidian',
+                                                    arr_dist=arr_dist)
+    selector.features = coords
+    selector.cluster()
+    selected_ids = selector.select()
+    labels_seleted = np.array([class_labels[elem] for elem in selected_ids])
+    for i in range(3):
+        assert len(labels_seleted[labels_seleted == i]) == 4
```

Comment:
I am not quite sure what the coding is checking here. I would assume we should check

- if  the selected label is correct
- if the number of selected data points matches the requirement

Reply:
I checked that from every cluster were selected equal amount of points(molecules). So, there are 3 clusters and 12 molecules needs to be selected. In this test, in particular, it is checked if 4 points were selected from every cluster.